Apple/__main__.py

```diff
@@ -0,0 +1,990 @@
+        ]:
+            step(context)
+
+    if host == "build":
+        hosts = []
+    elif host in {"all", "hosts"}:
+        hosts = all_host_triples(context.platform)
+    else:
+        hosts = [host]
+
+    for step_host in hosts:
+        for step in [
+            configure_host_python,
+            make_host_python,
+        ]:
+            step(context, host=step_host)
+
+    if host in {"all", "hosts"}:
+        package(context)
+
+
+def test(context: argparse.Namespace, host: str | None = None) -> None:
+    """The implementation of the "test" command."""
+    if host is None:
+        host = context.host
+
+    if context.clean:
+        clean(context, "test")
+
+    with group(f"Test {'XCframework' if host in {'all', 'hosts'} else host}"):
+        timestamp = str(time.time_ns())[:-6]
+        testbed_dir = (
+            CROSS_BUILD_DIR / f"{context.platform}-testbed.{timestamp}"
+        )
+        if host in {"all", "hosts"}:
+            framework_path = (
+                CROSS_BUILD_DIR / context.platform / "Python.xcframework"
+            )
+        else:
+            build_arch = platform.machine()
+            host_arch = host.split("-")[0]
+
+            if not host.endswith("-simulator"):
+                print("Skipping test suite non-simulator build.")
+                return
+            elif build_arch != host_arch:
+                print(
+                    f"Skipping test suite for an {host_arch} build "
+                    f"on an {build_arch} machine."
+                )
+                return
+            else:
+                framework_path = (
+                    CROSS_BUILD_DIR
+                    / host
+                    / f"Apple/{context.platform}"
+                    / f"Frameworks/{apple_multiarch(host)}"
+                )
+
+        run(
+            [
+                sys.executable,
+                "Apple/testbed",
+                "clone",
+                "--platform",
+                context.platform,
+                "--framework",
+                framework_path,
+                testbed_dir,
+            ]
+        )
+
+        run(
+            [
+                sys.executable,
+                testbed_dir,
+                "run",
+                "--verbose",
+            ]
+            + (
+                ["--simulator", str(context.simulator)]
+                if context.simulator
+                else []
+            )
+            + [
+                "--",
+                "test",
+                "--slow-ci" if context.slow else "--fast-ci",
+                "--single-process",
+                "--no-randomize",
+                # Timeout handling requires subprocesses; explicitly setting
+                # the timeout to -1 disables the faulthandler.
+                "--timeout=-1",
+                # Adding Python options requires the use of a subprocess to
+                # start a new Python interpreter.
+                "--dont-add-python-opts",
```

Comment:
See https://github.com/python/cpython/pull/138805#discussion_r2358537804.

Reply:
Sure - that makes sense. I've got some ideas for how this could be managed in the context of the iOS testbed - we can read options out of the Info.plist file to separate test suite options from interpreter options.

Reply:
I found it simpler not to separate them at all, but to let `PyConfig_SetBytesArgv` handle both of them at once, just like in a normal Python executable.